eZ/Bundle/EzPublishCoreBundle/Console/Application.php

```diff
@@ -19,11 +23,44 @@
  */
 class Application extends BaseApplication
 {
+    // See doRun()
+    private $kernel;
+
     public function __construct(KernelInterface $kernel)
     {
+        $this->kernel = $kernel;
         parent::__construct($kernel);
         $this->getDefinition()->addOption(
             new InputOption('--siteaccess', null, InputOption::VALUE_OPTIONAL, 'SiteAccess to use for operations. If not provided, default siteaccess will be used')
         );
     }
+
+    public function doRun(InputInterface $input, OutputInterface $output)
+    {
+        // boot() will be re-executed by parent, but kernel only boots once regardlessly
+        // @todo Contribute a console.init event to Symfony 4 in order to rather use that in v3
+        $this->kernel->boot();
+
+        $container = $this->kernel->getContainer();
+        $siteAccess = $container->get('ezpublish.siteaccess');
```

Comment:
Unfortunately this is not working with Symfony 4. At this point container is ready, thus not accepting private services.
```
The "ezpublish.siteaccess" service or alias has been removed or inlined when the container was compiled. You should either make it public, or stop using the container directly and use d  
  ependency injection instead.
```

Reply:
So, argument for doing event based then, ConsoleCommandListener has siteaccess injected.

Reply:
But we don't even need custom `Application` class, right? This seems to be much better approach:

```php
    public function onConsoleCommand(ConsoleCommandEvent $event)
    {
        $inputDefinition = $event->getCommand()->getApplication()->getDefinition();

        $inputDefinition->addOption(
            new InputOption('--siteaccess', null, InputOption::VALUE_OPTIONAL, 'SiteAccess to use for operations. If not provided, default siteaccess will be used')
        );
    }
```

Reply:
Console is loaded before ConsoleCommandEvent is ran, this is the issue I'm trying to solve.

Reply:
Approach updated.